contrib/gcppubsub/pkg/adapter/adapter.go

```diff
@@ -85,9 +89,15 @@ func (a *Adapter) receiveMessage(ctx context.Context, m PubSubMessage) {
 	}
 }
 func (a *Adapter) postMessage(ctx context.Context, logger *zap.SugaredLogger, m PubSubMessage) error {
+	et := eventType
+	if override, ok := m.Message().Attributes[eventTypeOverrideKey]; ok {
+		et = override
+		logger.Infof("overriding the cloud event type with %q", et)
+	}
```

Comment:
not sure I follow. as a first cut I simply want to be able to distinguish between different types that are coming through the pubsub. without this change it's impossible to understand what's on the wire if it goes through pubsub, later on we might want to do something more clever, but for now, this allows us to use pubsub as a generic ingress, consuming function can then look for cloud events of type (say, google.gcs) instead of only google.pubsub.topic.publish.
So, no unwrapping is necessary, we don't transform the event, just make it so that GCP PubSub source can be used as an ingress. Does that make sense?

Reply:
Sorry for the confusion! I was asking if the pubsub source wraps the original event beyond changing the type. If so that wrapping would need to be unwrapped here. Seems like no. :smile:

Reply:
no worries, yeah, no it leaves it untouched, just changes the cloud event type, that's all :)